### PR TITLE
GEOMESA-2856 Update NiFi docs

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/user/nifi.rst [hunk 1]
```diff
@@ -34,24 +34,44 @@ Install the Processors
 ~~~~~~~~~~~~~~~~~~~~~~
 
 To install the GeoMesa processors you will need to copy the nar files into the ``lib`` directory of your
-NiFi installation. There are currently three nar files:
+NiFi installation. There currently two common nar files, and seven datastore-specific nar files.
```
Comment (jnh5y) [inline docs/user/nifi.rst:37]: Is it possible to combine these common nars into one?

Comment (elahrvivaz) [reply docs/user/nifi.rst:37]: no, see https://nifi.apache.org/docs/nifi-docs/html/developer-guide.html#nars for details

File: docs/user/datastores/index_basics.rst [hunk 2]
```diff
@@ -64,6 +64,38 @@ To index an attribute, add an ``index`` key to the attribute descriptor user dat
 
 Setting the user data can be done in multiple ways. See :ref:`set_sft_options` for more details.
 
+Attribute indices also support a secondary, tiered index structure. This can improve attribute queries
+that also contain a spatial and/or temporal predicate. Unless configured differently, the default geometry
+and date attributes will be used to create a secondary Z3 of XZ3 index.
```
Comment (jnh5y) [inline docs/user/datastores/index_basics.rst:69]: Do you mean Z3 or XZ3?  

Also, referring to this a 'secondary' index is slightly confusing.

Comment (jnh5y) [reply docs/user/datastores/index_basics.rst:69]: Oh, I'm just confused.

Comment (elahrvivaz) [reply docs/user/datastores/index_basics.rst:69]: oh, yes it should be 'or', good catch.

**Pull request closed without merging**